warehouse/attestations/services.py

```diff
@@ -36,53 +33,17 @@
 from warehouse.attestations.interfaces import IIntegrityService
 from warehouse.attestations.models import Provenance as DatabaseProvenance
 from warehouse.metrics.interfaces import IMetricsService
-from warehouse.oidc.models import (
-    GitHubPublisher as GitHubOIDCPublisher,
-    GitLabPublisher as GitLabOIDCPublisher,
-    OIDCPublisher,
-)
 from warehouse.utils.exceptions import InsecureIntegrityServiceWarning
 
 if typing.TYPE_CHECKING:
     from warehouse.packaging.models import File
 
 
-def _publisher_from_oidc_publisher(publisher: OIDCPublisher) -> Publisher:
-    """
-    Convert an OIDCPublisher object in a pypi-attestations Publisher.
-    """
-    match publisher.publisher_name:
-        case "GitLab":
-            publisher = typing.cast(GitLabOIDCPublisher, publisher)
-            return GitLabPublisher(
-                repository=publisher.project_path, environment=publisher.environment
-            )
-        case "GitHub":
-            publisher = typing.cast(GitHubOIDCPublisher, publisher)
-            return GitHubPublisher(
-                repository=publisher.repository,
-                workflow=publisher.workflow_filename,
-                environment=publisher.environment,
-            )
-        case _:
-            raise AttestationUploadError(
-                f"Unsupported publisher: {publisher.publisher_name}"
-            )
-
-
 def _build_provenance(
     request: Request, file: File, attestations: list[Attestation]
 ) -> DatabaseProvenance:
-    try:
-        publisher: Publisher = _publisher_from_oidc_publisher(request.oidc_publisher)
-    except AttestationUploadError as exc:
-        sentry_sdk.capture_message(
-            f"Unsupported OIDCPublisher found {request.oidc_publisher.publisher_name}"
-        )
-        raise exc
-
     attestation_bundle = AttestationBundle(
-        publisher=publisher,
+        publisher=request.oidc_publisher.attestation_identity,
```

Comment:
Bit of a nit: This function is now simple enough that I think it can be eliminated and we can call this inline instead.

Reply:
Yeah, good call -- doing!